### PR TITLE
Mobile-first approach to grids (and get rid of o-useragent)

### PR DESCRIPTION
**Pull request by kaelig**
The current implementation of grids has some great ideas but its main limitation is that it doesn't enable us to add layouts.

Adding a new layout size (for example: adding the [XS layout](https://github.com/Financial-Times/o-grid/issues/61)) would result in gaps where "no layout exists", that's where a mobile-first approach comes in to solve this problem makes design changes easier. 

### Files to review

- [src/sass/_variables.scss](https://github.com/Financial-Times/o-grid/blob/refactor-mqs/src/scss/_variables.scss)
- [src/sass/_main.scss](https://github.com/Financial-Times/o-grid/blob/refactor-mqs/src/scss/_main.scss)
- [src/js/getCurrentLayout.js](https://github.com/Financial-Times/o-grid/blob/refactor-mqs/src/js/getCurrentLayout.js): new JS helper to know what layout is currently displayed to the user

### Have a look
- [Test demo](http://kaelig.fr/ft/origami/o-grid/demos/test.html)
- [All demos](http://kaelig.fr/ft/origami/o-grid/demos/)

### Documentation
- [New README](https://github.com/Financial-Times/o-grid/blob/refactor-mqs/README.md)
- [SassDoc](http://kaelig.fr/ft/origami/o-grid/docs/sass/)

### Sass MQ

A Sass mixin that helps manipulating media queries in an elegant, mobile-first way. 

It mainly helps with breakpoint management and could also help with IE 8 Support ([read about how the Guardian uses it](http://www.theguardian.com/info/developer-blog/2013/oct/14/mobile-first-responsive-ie8)).

- [Documentation](https://github.com/sass-mq/sass-mq)
- [SassDoc](http://sass-mq.github.io/sass-mq/)

Let's have a chat about this very soon :) We'd like to ship it in early January and update this list of dependents as soon as possible: next-header o-ft-footer o-ft-forms o-ft-header o-hierarchical-nav o-techdocs.

### CODE REVIEW

**Comment by kaelig** [comment]
Good point, @wheresrhys. Let's talk about it on Thursday.

**Comment by matthew-andrews** [comment]
Hi Kaelig just had another go to try to get this to work in Next.

We previously used `extend` quite a lot to make some of our elements behave as if they had had grid data attributes.
e.g. https://github.com/Financial-Times/next-header/blob/master/src/scss/_base.scss#L11-L13

I had assumed we'd be able to do something similar with mixins, e.g.:-
```scss
@include @oGridRow();
@include @oGridColumn();
@include @oGridColspan(12);
```

But that doesn't seem to be the case…  Should we continue expect to do something like this?  It's quite nice to keep component's markup simple and not need to have to put the data attributes everywhere.  But I can see that doing this makes thing simpler in other ways.

Basically, I've no strong opinions either way but I would like some guidance as to which approach we should move towards.

**Comment by kaelig** [comment]
@matthew-andrews looks like I need to document this further, because what you describe is entirely doable but I hadn't thought of putting examples in the readme…

```scss
// equivalent to data-o-grid-colspan="8 S5 M9"
.el {
  @include oGridColumnWidth(8);

  @include oGridRespondTo(S) {
   @include oGridColumnWidth(5);
  }
  @include oGridRespondTo(M) {
   @include oGridColumnWidth(9);
  }
}
```

You could also use the `oGridRespondTo` mixin like this:

```scss
.el {
  @include oGridRespondTo($until: L) {
   @include oGridColumnWidth(9); // 9 col wide until L
  }
  @include oGridRespondTo($from: S, $until: L) {
   @include oGridColumnWidth(9); // 9 col wide from S until L
  }
  @include oGridRespondTo($from: M, $until: L) {
   @include oGridColumnWidth(9); // 9 col wide only on M layout
  }
}
```

Does that help? Let me know if you see additional use cases that I should document, and I'll add them to the README.
